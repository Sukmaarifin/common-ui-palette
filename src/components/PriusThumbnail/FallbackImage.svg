<svg width="87" height="87" viewBox="0 0 87 87" fill="none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<rect width="87" height="87" fill="url(#pattern0)"/>
<defs>
<pattern id="pattern0" patternContentUnits="objectBoundingBox" width="1" height="1">
<use xlink:href="#image0" transform="scale(0.0114943)"/>
</pattern>
<image id="image0" width="87" height="87" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAFcAAABXCAYAAABxyNlsAAALGklEQVR4Ae3d6ZLUyBWG4aoGmn1YGm7Pl+Cf9j3439zCXB1E04ZmXxra8Sj84TNplUqllgCHlREipVSe7c2TR6oqZtheXl4en56e/uPy8vIvl5eXjzZruxKB7Xb7crvd/vH06dO/bZ89e/b78fHxX3/77bfNtWvXrqR4Fd5svn79unn9+vXm4uLi9+3z58//eXJy8mgFO19qAHx2dvbySClYwc4HliY8cT2aV+2qrRJY4VYaM5+vcGcGWtWtcCuNmc9XuDMDrepWuJXGzOcr3JmBVnUr3Epj5vMV7sxAq7oVbqUx8/kKd2agVd0Kt9KY+XyFOzPQqm6FW2nMfL7CnRloVXe9Xvwq59++fdt8/vx5o79z506vWxcXF5uPHz92351ev35949hut71zf9bgLwE3ML98+dJB9U3+8fGxn0o6eDdv3vwvPm/fvt0cHR11C/D+/fvu5xXXN27c6EAH+M/8IeCnwA1M2QloYALqt7xk4bt377r7fXCBf/To0fff/S4vLzs9xh0fPnzoerYCuvYWYun2Q+AOwbTtBd3XZKEMbRuQdFZASkLg1fnmWTyLCLpSoq/zI6efs7T0R1W9m3A+BSZggSZA2xlcIIzXoI1VEMBlDuDm5nDtoKs2PtLjIJ9MN5fuWl6mlpZZ4FaYeRDZ4o5dmVlhknfUFkiCpLOWhsA1nx6ZmVbPK+Ccp6c/PkY2ugI9wPlWszvndAy1SXCXgNk6mUwEQGYNwW1lc10XMGN6gLN47XnA1fnitWjJ8pSWW7dudc+IOreej4LbB1OwsmpqZlYn+s7ZBICNtu4KMrDN29eAqVs7GdrKBXj6muXx5fbt252Y3eSBO9QG4VohgQlgaZitk4EGLpjJZPNcyzDN+FADwfYGxS4YarG5q7QA7NCqD7t0DsL99OlTl5l9L/J1y3Eqju0y1Deemmbh2sDpC9BadzOeIPfZdd9CKC2tjfjEjljZqRnufo3TtVKgDenrJmw2m8GKnKzJZI5aMdnAGb3rfQFGvvacJvfkyZNOT72Xc3O01F3n7PFLS+DdxcAf5M0l2zbjPoSIxTG2VT92yQzCrUFRQKFjCszWAfVMU7fo69OZsWRufBhbEmKTLR9O7JS65ek3ZhfI2OiNXNtnt1gQevbObxXUa8I18ACpc5IR+n2NQzJeprx58+b7u6ut1pdVFa77sZWgxtg0h98S5d69e51tuvI8sXDAj4GV+M1ty0df7IM1l0CyBgArR3FtQDFkS7W12dwcAjJPkB4u9MZZ9/y1S/dqC1zzAE0ZyhN7DFz6Mo8cXR7S4rl//37nk91T/ak+1PNkLn+zwPV+e74XroAFFbitAo4/fPjQX5nsQALCeIVJtjpPJnPo52gAtPrpE5SHnodIDSzwW5l6TZZMGl8ctcnivNrV8fa8JsNscNUljXJHBSEbgRWALQ/W3bt3/wQzTpoja0DiHOCvXr3qFoeevu0WuOYqJWAlyOpHbLQ9ubxOJvPqHD6zwf6+FrviyO4ZktmbuSAwniA5WEuDa3MeP378p6A5EIjZ7ra+ulez2LnFM4dMGyS7mnnuJ8OAHQOXf3S/fPmyA8JXh3E78vz8vCtnAbcLlvuZww869rX9M/4dGFACa+EywhjDoMuuClU9A4GsOQEdx1yr2x4qsqhtgUsW4LQxYDOXLB/pVwLir/u7PmFGNj0dWhKtTYLMq/0ouABwLHCrArA5q+VB8eDBg++rbPvIjppxcRQgwdZFoS9zYye7xriF08bCpVuG0ptdQ96CimlMBppPXhPraJlOYs8fMiZBARM4xHKeFQUgY+5zKiA8SICWzWrtixcvOrgWQJ32gYKeLFbcMqbxIyUpY5nT17Or5IDqvEKRMK2dPh0ZS0yHwB2VuQlKQGA5EiTjnJYdesazfQVkUcwFM+WCDJjmx+kEYZsCUkEEJL300+vY1/hk+5Ljc7VFPx2JaZ8u8trscCm10uDIzD64thhgKQGcCEzAyPfBTFCCBMNBl5ZXpsAFByh6Mxb5tnefnpOTkw6q7G2b99yxLQsjrvi1T3ZU5lKyr+4KJNnLkV2ZGYcqTLA04NghK9NrM9+iWly2zB1qKQd58PQBUY7o2rcLApa9RTJXMBzWGHPEKVA5KjuqIzX4CtN88wKTXLZdZKI/+gKXjDeSIbjRz6d9rd2FffPjGx/4k+u+uXVsdOYKRu1MkNUpkGVGQDBwKMzqlHMZLEsCkT7NtXE2q73Im+cNpL53515fL6ZdujI/dg7JWrKj4Zos4L66a+vZXhyVNckcIMj0ZWYc39WT87rUwhWoMbbqQy96gE1JythQzwZbfboil0xdHC6IeajFOOBW3zEVZnSlD8Bc67Nr2GezBWJR+dB+gVR1tOf83QetZm4Wu9XTd31Q5lLcV3cZ9+Q9xHCfM8Ak88FLKcjcwK1+1Huy1l8UCYzcG+rpkjBDLfr4dMjCHQy3r+7KIA4eCrfCBBU8JYYNW1swtQU2O+aQT+DAeoAd6kOfrmozJYEtNvP2UefsOj8ILiW2kaBTGmJQcPvaLpgWxwMxjlsoQdmu7KQFLqCBQpY//PDV56GNHUcWtpXP4iXOXLfz+q4Phiuotu5yDriaSYwNwZRl5PoayLtet0Agl1ppbkrV6elpt22HXgn77FVd7f34uK8ut3KuD4bLEV8dalbRASInZFAFmmyQXUMwW8fM3/XpqcK1W2S8Opis972FLPMtm7mulRz+KTV9HyYkTBao9SWZ+kPgZjsmSE4LBhDfmYLvWsBZ9dbhOa7jh8VkOw1AoGQxMPzhC38lhfkWWm8e8Obxua9VuOQOaf/x6gApDrd1V7DJIBkVpw5Qu3OqRaqHifSzCVqyNuP8AK/6YA4dyo1sjlySwbWjJgT56PghmSsAQVnxPNRCRQAcF1gbXOaM6SvIGmwrCww7/NCDk76d65p/yo0sjQ3j4pH9ARnZXAd8XcTMGeonZ+6uupvaOgSldSiBpm/v77oG1xf0eYC1NgO69nTVMhLdfWPRNyVrOztRfkhvpa1+thEnUrP6nGx1B2L69v7Ya34IHLxkWZU11mabuTmS6bmuss6j84fCZbiv7rbO5ToQ02f8qr3gLab6z58xjUygVfABHOB08Vf74XB31d04FZBxsPNygT9AVf/Hwt3lQqD3+Qtu3yvcLl0Zn1RzCQum1t0E1+dcjC3Rs6vuLtmmZm7/R6QRnta6a3oydYTorFOUBcEv1ZQJWT0laSZnrmBkTd53lwqu1as2sunwMRzY7Jp27hzXU7OW7SvBrXV3jkD6dASmuuoAVSbJWFC9t455Q+nTPWbsp8EVnE88c7YWZoJjK79oeO0z7lD3XXvyg5zDwk/Zym0sbNA1pV05cxnP++4UB/bBFFhenaIfwPpVpPHA1vvOIN8jBHbtD4Fupxz6ncJ3P3MytRc8B9pgd+mbAnOXrjoeeHUstgLedx7OU1bI8F8v89tFJG9XuD+lTZMqlmzXfM9Qhr+fJsDUTMFxNtu8LzO/C1/xBCx2HLXVsuKhmK8p29JCXpa30KuuofNZ4Na6+zNhDgVa74HoaHdbMlyvtOinlgT2rgxX5nHCi7zsdG7M4QG0ZGZWYHOc21FTS0Cf/SvDpTT/wcb/Gsw+IHOOzQJ3yufuOYP4VXVN/vj7qwb0K/m1wl1wNVa4K9wFCSyoes3cFe6CBBZUvWbuCndBAguqXjN3hbsggQVVr5m7wl2QwIKq18xd4S5IYEHVR/7NRD97rG0+AnjiCu4f+Xl6PvX/v5qAxRPX6+fn53/3P584Oztb//XUGXKi/uup/wKDh4XDsaOsdAAAAABJRU5ErkJggg=="/>
</defs>
</svg>
