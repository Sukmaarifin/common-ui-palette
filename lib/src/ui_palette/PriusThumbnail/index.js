import React, { useEffect, useState } from "react";
import clsx from "clsx";
import { makeStyles } from "@material-ui/core/styles";
import { camelize } from "../../helpers";
import FallbackImage from "./FallbackImage.svg";
var useStyles = makeStyles(function () { return ({
    container: {
        cursor: "pointer",
        border: "none",
        background: "transparent",
        "&:hover": {
            opacity: "0.6",
        },
        "&:focus, &:active": {
            outline: "none",
        },
    },
    image: {
        display: "block",
        objectFit: "cover",
    },
    small: {
        width: "2.5rem",
        height: "2.5rem",
    },
    medium: {
        width: "4rem",
        height: "4rem",
    },
    big: {
        width: "5rem",
        height: "5rem",
    },
}); });
var PriusThumbnail = function (_a) {
    var altText = _a.altText, className = _a.className, defaultSource = _a.defaultSource, imageSource = _a.imageSource, setSize = _a.setSize, size = _a.size, onError = _a.onError, onClick = _a.onClick, id = _a.id;
    var classes = useStyles();
    var _b = useState(), source = _b[0], setSource = _b[1];
    var handleDefaultSrc = function (event) {
        event.target.src = FallbackImage;
        if (onError)
            onError();
    };
    useEffect(function () {
        if (!Boolean(imageSource) && !Boolean(defaultSource)) {
            setSource(FallbackImage);
        }
        else if (Boolean(defaultSource)) {
            setSource(defaultSource);
        }
        else {
            setSource(imageSource);
        }
    }, [defaultSource, imageSource]);
    return onClick ? (React.createElement("button", { className: className ? className : classes.container, onClick: onClick, id: id && id + "-button-" + camelize(altText), "data-identity": id && id + "-button-" + camelize(altText) },
        React.createElement("img", { onError: handleDefaultSrc, alt: altText, style: {
                width: setSize,
                height: setSize,
            }, className: clsx(classes.image, classes[size]), src: source }))) : (React.createElement("img", { onError: handleDefaultSrc, alt: altText, className: className ? className : clsx(classes.image, classes[size]), src: source }));
};
export default PriusThumbnail;
