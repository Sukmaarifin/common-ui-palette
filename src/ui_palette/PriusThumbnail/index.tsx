import React, { useEffect, useState } from "react";
import clsx from "clsx";
import { makeStyles } from "@material-ui/core/styles";

import { camelize } from "../../helpers";

import FallbackImage from "./FallbackImage.svg";

const useStyles = makeStyles(() => ({
  container: {
    cursor: "pointer",
    border: "none",
    background: "transparent",
    "&:hover": {
      opacity: "0.6",
    },
    "&:focus, &:active": {
      outline: "none",
    },
  },
  image: {
    display: "block",
    objectFit: "cover",
  },
  small: {
    width: "2.5rem",
    height: "2.5rem",
  },
  medium: {
    width: "4rem",
    height: "4rem",
  },
  big: {
    width: "5rem",
    height: "5rem",
  },
}));

export type ThumbnailSize = "small" | "medium" | "big";

export type PriusThumbnailProps = {
  defaultSource?: React.ReactNode;
  altText: string;
  className?: string;
  imageSource: string;
  setSize?: string;
  size: ThumbnailSize;
  onError?: () => void;
  onClick?: (event: any) => void;
  id?: string;
};

const PriusThumbnail = ({
  altText,
  className,
  defaultSource,
  imageSource,
  setSize,
  size,
  onError,
  onClick,
  id,
}: PriusThumbnailProps) => {
  const classes = useStyles();

  const [source, setSource] = useState<any>();

  const handleDefaultSrc = (event: any) => {
    event.target.src = FallbackImage;
    if (onError) onError();
  };

  useEffect(() => {
    if (!Boolean(imageSource) && !Boolean(defaultSource)) {
      setSource(FallbackImage);
    } else if (Boolean(defaultSource)) {
      setSource(defaultSource);
    } else {
      setSource(imageSource);
    }
  }, [defaultSource, imageSource]);

  return onClick ? (
    <button
      className={className ? className : classes.container}
      onClick={onClick}
      id={id && `${id}-button-${camelize(altText)}`}
      data-identity={id && `${id}-button-${camelize(altText)}`}
    >
      <img
        onError={handleDefaultSrc}
        alt={altText}
        style={{
          width: setSize,
          height: setSize,
        }}
        className={clsx(classes.image, classes[size])}
        src={source}
      />
    </button>
  ) : (
    <img
      onError={handleDefaultSrc}
      alt={altText}
      className={className ? className : clsx(classes.image, classes[size])}
      src={source}
    />
  );
};

export default PriusThumbnail;
