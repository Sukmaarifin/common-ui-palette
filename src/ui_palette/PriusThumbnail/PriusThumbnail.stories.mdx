import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import Grid from "@material-ui/core/Grid";

import PriusThumbnail from "../PriusThumbnail";

<Meta
  component={PriusThumbnail}
  id="Components|PriusThumbnail"
  title="Components|PriusThumbnail"
/>

# PriusThumbnail

`PriusThumbnail` is a simple preview image and used as a placeholder if source
of the image is not found. It provide with diferent size options.

<Preview>
  <Story name="Basic">
    <Grid container spacing={0}>
      <Grid item xs={2}>
        <PriusThumbnail altText="alt-text" size="big" />
      </Grid>
      <Grid item xs={2}>
        <PriusThumbnail altText="alt-text" size="medium" />
      </Grid>
      <Grid item xs={2}>
        <PriusThumbnail altText="alt-text" size="small" />
      </Grid>
    </Grid>
  </Story>
</Preview>

## Props

<Props of={PriusThumbnail} />
