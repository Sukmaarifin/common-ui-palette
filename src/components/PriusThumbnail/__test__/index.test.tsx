import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import PriusThumbnail, { PriusThumbnailProps } from '..';

describe('PriusThumbnail test', () => {
  const props: PriusThumbnailProps = {
    imageSource: '#some-cool-image.jpg',
    altText: 'Indonesia',
    size: 'small',
  };

  // snapshot
  it('renders correctly', () => {
    const wrapper = mount(<PriusThumbnail {...props} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
