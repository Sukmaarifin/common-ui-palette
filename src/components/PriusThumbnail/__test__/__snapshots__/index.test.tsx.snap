// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriusThumbnail test renders correctly 1`] = `
<PriusThumbnail
  altText="Indonesia"
  imageSource="#some-cool-image.jpg"
  size="small"
>
  <img
    alt="Indonesia"
    className="makeStyles-image-2 makeStyles-small-3"
    onError={[Function]}
    src="#some-cool-image.jpg"
  />
</PriusThumbnail>
`;
